jobs/opi/templates/bpm.yml.erb

```diff
@@ -5,9 +5,11 @@ processes:
     - connect
     - --config=/var/vcap/jobs/opi/config/opi.yml
     env:
-      KUBERNETES_SERVICE_HOST: "<%= p("opi.kube_service_host") %>"
-      KUBERNETES_SERVICE_PORT: "<%= p("opi.kube_service_port") %>"
-    <% if properties.opi&.k8s&.host_url.nil? %>
+      <% host = p("opi.kube_service_host", "") %>
+      <%= "KUBERNETES_SERVICE_HOST: \"#{host}\"" unless host.empty? %>
+      <% port = p("opi.kube_service_port", "") %>
+      <%= "KUBERNETES_SERVICE_PORT: \"#{port}\"" unless port.empty? %>
+    <% unless p("opi.k8s.host_url", "").empty? %>
```

Comment:
The condition is reversed, it used to render the block below when the host_url was empty, now it does the opposite. The check is there to only mount the needed volumes for inCluster communication with the Kube api when we are actually "in cluster". `host_url` is used to point to a Kubernetes cluster when you are in the Bosh world in which case you don't need to mount the additional volumes. A lot of logic hidden in one line I'm afraid.

Reply:
You are right. How about we introduce the `opi.k8s.mount_service_account` property defaulting to `true`?

Reply:
We based the condition on the `host_url` to avoid breaking things for the teams working with Bosh (they set `host_url`). A new property sounds good but we need to notify people about this breaking change. cc @jenspinney @jspawar

Reply:
Let's go with a new property - it makes more sense and it's still flexible enough to support both scenarios.

Reply:
Yeah, I don't see any problems with this, but we (the team that Jwal and I are on) aren't actively using or testing against eirini-bosh-release anymore. The CAPI (and networking) teams are though. To be able to continue using eirini-bosh-release deployed by BOSH, I imagine https://github.com/cloudfoundry-community/eirini-bosh-release/blob/master/operations/add-eirini.yml will have to be updated to set the new property to false. Could you also make that change at the same time you introduce the new property, @f0rmiga?

Tagging @selzoc @cwlbraa here since they're actively using eirini-bosh-release with BOSH.